### PR TITLE
subscribe to local and remote changes

### PR DESCRIPTION
**Pull request by tshedor**

### CODE REVIEW

File: packages/brick_offline_first_with_graphql/lib/src/offline_first_with_graphql_repository.dart [hunk 1]
```diff
@@ -109,32 +144,78 @@ abstract class OfflineFirstWithGraphqlRepository
...
+  @visibleForTesting
+  Future<void>
+      notifySubscriptionsWithLocalData<_Model extends OfflineFirstWithGraphqlModel>() async {
+    final queriesControllers = subscriptions[_Model]?.entries;
+    if (queriesControllers?.isEmpty ?? true) return;
+
+    for (final queryController in queriesControllers!) {
+      final query = queryController.key;
+      final controller = queryController.value;
+      if (controller.isClosed || controller.isPaused) continue;
```
Comment (mateominato) [inline packages/brick_offline_first_with_graphql/lib/src/offline_first_with_graphql_repository.dart:158]: what does this do?

**Review by mateominato — APPROVED** [review]
wild stuff

File: packages/brick_offline_first_with_graphql/lib/src/offline_first_with_graphql_repository.dart (hunk 1 above)
Comment (tshedor) [reply packages/brick_offline_first_with_graphql/lib/src/offline_first_with_graphql_repository.dart:158]: iterates through all the subscribed controllers and fires the queries that have been cached for each `subscribe` invocation

**Comment by Josiassejod1** [comment]
this is awesome stuff

**Pull request merged**